README.md

```diff
@@ -40,6 +40,27 @@ The following command will restart all pods in the `web` and `jobs` deployments:
 
 `kubernetes-restart <kube namespace> <kube context> --deployments=web,jobs`
 
+### Running one off tasks
+
+To trigger a single use job such as a rake task, use the following command:
+
+`kubernetes-run <kube namespace> <kube context> --command=rake <arguments>`
+
+This command assumes that there is a `PodTemplate` named `task-runner-template`, which contains a full pod specification in it's `data`. The pod specification in turn must have a container named `task-runner`. This container will be passed the arguments and command passed to `kubernetes-run`.
+
+#### Creating a PodTemplate
+
+The `PodTemplate` object should have a field `template` containing a `Pod` specification which does not include the `apiVersion` or `kind` parameters. An example is provided in this repo in `template-runner.yml`. 
+
+#### Providing multiple different task-runner configurations
+
+If your application has multiple different formats of task runners that are needed you can specify which template to use by using the `--template` option. This template is still expected to provide a container called `task-runner`.
```

Comment:
I just figured that it is less confusing to have a single rule for which container is used... That way if you start with one container and add a sidecar, it won't magically break because you didn't remember to rename to container correctly.